src/Build.UnitTests/Construction/SolutionProjectGenerator_Tests.cs

```diff
@@ -2640,20 +2734,23 @@ public void DirectorySolutionPropsTest(string projectName, bool enable)
         /// Regression test for https://github.com/dotnet/msbuild/issues/6236
         /// </summary>
         [Theory]
-        [InlineData("http://localhost:8080")]
-        [InlineData("a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-a-really-long-string-")]
-        public void AbsolutePathWorksForUnsupportedPaths(string relativePath)
+        [InlineData("http://localhost:8080", false)]
+        [InlineData("http://localhost:8080", true)]
+        [InlineData(_longLineString, false)]
+        [InlineData(_longLineString, true)]
+        public void AbsolutePathWorksForUnsupportedPaths(string relativePath, bool useNewParser)
         {
```

Comment:
to answer your question, Wave17_14 is enabled in the whole build. Tests are modified so both old and new parsers are tested:
- `SolutionProjectGenerator_Tests` are modified to use both old and new parsers. 
- `SolutionFile_Tests` that were old parser specific are moved to `SolutionFile_OldParser_Tests` that use old parser. `SolutionFile_Tests` uses `Parse()` method which calls new parser because Wave17_14 is enabled.